shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/connection/JDBCBackendConnection.java

```diff
@@ -176,43 +176,48 @@ public void unmarkResourceInUse(final JDBCDatabaseCommunicationEngine databaseCo
     
     @Override
     public Void prepareForTaskExecution() throws BackendConnectionException {
-        if (!connectionSession.getTransactionStatus().isInConnectionHeldTransaction()) {
-            connectionStatus.waitUntilConnectionRelease();
-            connectionStatus.switchToUsing();
-        }
-        if (!connectionSession.isAutoCommit() && !connectionSession.getTransactionStatus().isInTransaction()) {
-            JDBCBackendTransactionManager transactionManager = new JDBCBackendTransactionManager(this);
-            try {
-                transactionManager.begin();
-            } catch (SQLException ex) {
-                throw new BackendConnectionException(ex);
+        synchronized (this) {
+            connectionReferenceCount++;
+            if (!connectionSession.isAutoCommit() && !connectionSession.getTransactionStatus().isInTransaction()) {
+                JDBCBackendTransactionManager transactionManager = new JDBCBackendTransactionManager(this);
+                try {
+                    transactionManager.begin();
+                } catch (SQLException ex) {
+                    throw new BackendConnectionException(ex);
+                }
             }
+            return null;
         }
-        return null;
     }
     
     @Override
     public Void closeExecutionResources() throws BackendConnectionException {
-        Collection<Exception> result = new LinkedList<>();
-        result.addAll(closeDatabaseCommunicationEngines(false));
-        result.addAll(closeFederationExecutor());
-        if (!connectionSession.getTransactionStatus().isInConnectionHeldTransaction()) {
-            result.addAll(closeDatabaseCommunicationEngines(true));
-            result.addAll(closeConnections(false));
-            connectionStatus.switchToReleased();
-        }
-        if (result.isEmpty()) {
-            return null;
+        synchronized (this) {
+            if (connectionReferenceCount > 0 && connectionReferenceCount-- > 1) {
```

Comment:
Maybe `--connectionReferenceCount` ? or why not use `connectionReferenceCount > 1`?

Reply:
> Maybe `--connectionReferenceCount` ? or why not use `connectionReferenceCount > 1`?

Count may not be increased before decrease.

Reply:
Got it.thanks